File: backend/src/HIS.Core/Entities/ObstetricRegister.cs
namespace HIS.Core.Entities;

/// <summary>
/// Sổ sinh đẻ (Birth register) — #154 F1.8. Register pháp lý khoa Sản:
/// mỗi record = 1 ca sinh. Nhập tay (không chỉ đếm trong tiền sử). Soft-delete IsDeleted.
/// </summary>
public class BirthRegister : BaseEntity
{
    public int RegisterNo { get; set; }            // số thứ tự sổ
    public DateTime DeliveryDate { get; set; }     // ngày giờ sinh

    // Sản phụ
    public string MotherName { get; set; } = string.Empty;
    public int MotherAge { get; set; }
    public string? MotherIdNumber { get; set; }    // CCCD/CMND
    public string? MotherAddress { get; set; }

    public int GestationalWeeks { get; set; }      // tuổi thai (tuần)
    public string? ParaInfo { get; set; }          // PARA / số lần sinh
    public string? DeliveryMethod { get; set; }    // Đẻ thường / Mổ / Forceps / Giác hút

    // Trẻ
    public int BabyCount { get; set; } = 1;        // số con (sinh đôi...)
    public int BabyGender { get; set; }            // 1=Nam, 2=Nữ, 0=chưa rõ
    public decimal BabyWeight { get; set; }        // gram
    public string? BabyCondition { get; set; }     // Sống / Chết / Dị tật

    public string? Attendant { get; set; }         // BS/NHS đỡ đẻ
    public string? Notes { get; set; }
}

/// <summary>
/// Sổ theo dõi nạo phá thai (Abortion register) — #154 F1.8. Register pháp lý khoa Sản:
/// mỗi record = 1 ca. Nhập tay. Soft-delete IsDeleted.
/// </summary>
public class AbortionRegister : BaseEntity
{
    public int RegisterNo { get; set; }            // số thứ tự sổ
    public DateTime ProcedureDate { get; set; }    // ngày thực hiện

    public string PatientName { get; set; } = string.Empty;
    public int PatientAge { get; set; }
    public string? PatientIdNumber { get; set; }   // CCCD/CMND
    public string? PatientAddress { get; set; }

    public int GestationalWeeks { get; set; }      // tuổi thai (tuần)
    public string? Method { get; set; }            // Hút / Nạo / Thuốc
    public string? Reason { get; set; }            // lý do
    public string? Performer { get; set; }         // người thực hiện
    public string? Complications { get; set; }     // tai biến / biến chứng
    public string? Notes { get; set; }
}

File: frontend/src/api/obstetricRegister.ts
import apiClient from '../services/apiClient';

const BASE = '/obstetric-register';

// ─── DTO interfaces ─────────────────────────────────────────────────────────

export interface BirthRegisterDto {
  id: string;
  registerNo: number;
  deliveryDate: string;       // ISO
  motherName: string;
  motherAge: number;
  motherIdNumber?: string;
  motherAddress?: string;
  gestationalWeeks: number;
  paraInfo?: string;
  deliveryMethod?: string;
  babyCount: number;
  babyGender: number;         // 1=Nam, 2=Nữ
  babyWeight: number;         // gram
  babyCondition?: string;
  attendant?: string;
  notes?: string;
}

export interface AbortionRegisterDto {
  id: string;
  registerNo: number;
  procedureDate: string;      // ISO
  patientName: string;
  patientAge: number;
  patientIdNumber?: string;
  patientAddress?: string;
  gestationalWeeks: number;
  method?: string;
  reason?: string;
  performer?: string;
  complications?: string;
  notes?: string;
}

export interface ObstetricReportDto {
  from: string;
  to: string;
  totalDeliveries: number;
  totalBabies: number;
  babiesAlive: number;
  babiesDead: number;
  maleBabies: number;
  femaleBabies: number;
  deliveriesByMethod: Record<string, number>;
  totalAbortions: number;
  abortionsByMethod: Record<string, number>;
  abortionComplications: number;
}

// ─── Sổ sinh đẻ ───────────────────────────────────────────────────────────────

export const getBirths = (from?: string, to?: string, keyword?: string) =>
  apiClient.get<BirthRegisterDto[]>(`${BASE}/births`, { params: { from, to, keyword } }).then(r => r.data);

export const saveBirth = (dto: Partial<BirthRegisterDto>) =>
  apiClient.post<BirthRegisterDto>(`${BASE}/births`, dto).then(r => r.data);

export const deleteBirth = (id: string) =>
  apiClient.delete(`${BASE}/births/${id}`).then(r => r.data);

// ─── Sổ theo dõi nạo phá thai ───────────────────────────────────────────────

export const getAbortions = (from?: string, to?: string, keyword?: string) =>
  apiClient.get<AbortionRegisterDto[]>(`${BASE}/abortions`, { params: { from, to, keyword } }).then(r => r.data);

export const saveAbortion = (dto: Partial<AbortionRegisterDto>) =>
  apiClient.post<AbortionRegisterDto>(`${BASE}/abortions`, dto).then(r => r.data);

export const deleteAbortion = (id: string) =>
  apiClient.delete(`${BASE}/abortions/${id}`).then(r => r.data);

// ─── Báo cáo BYT ──────────────────────────────────────────────────────────────

export const getObstetricReport = (from: string, to: string) =>
  apiClient.get<ObstetricReportDto>(`${BASE}/report`, { params: { from, to } }).then(r => r.data);
